src/goose/_logger.py

```diff
@@ -0,0 +1,20 @@
+import logging
+from pathlib import Path
+
+_LOGGER_NAME = "goose"
+_LOGGER_FILE_NAME = "goose.log"
+
+
+def setup_logging(log_file_directory: Path, level: int = logging.INFO) -> None:
+    logger = logging.getLogger(_LOGGER_NAME)
+    logger.setLevel(level)
+    log_file_directory.mkdir(parents=True, exist_ok=True)
+    file_handler = logging.FileHandler(log_file_directory / _LOGGER_FILE_NAME)
```

Comment:
what do you think about bucketing the logs by time (e.g. `yyyy-mm-dd-goose-cost.log`)? not sure how verbose the logging could be, but we sorta also get rolling logs for free too

Reply:
Yeah, definitely we can use the TimedRotatingFileHandler when the log gets verbose.  At the moment we only log the cost, but I think it will be helpful to log errors.

Reply:
makes sense! if we wanna go long-term we could also go one step further extending `logging.Filter` to some intelligent routing [or segment] (e.g. `api-costs.log` vs `goose.log`)

Reply:
yeah, like tags